src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -25,15 +27,20 @@ def __init__(self, pex_path, interpreter):
   def pex(self):
     return self._pex
 
-  def run(self, workunit_factory, args, **kwargs):
-    cmdline = ' '.join(self._pex.cmdline(args))
+  @contextmanager
+  def run_with(self, workunit_factory, args, **kwargs):
```

Comment:
When I try to make this change, I see:
```
  File "/Users/dmcclanahan/tools/pants/src/python/pants/backend/native/tasks/conan_fetch.py", line 137, in execute_codegen
    conan_install_stdout = workunit.output('stdout').read_from(0)
  File "/Users/dmcclanahan/tools/pants/src/python/pants/util/rwbuf.py", line 32, in read_from
    self._io.seek(pos)

Exception message: seek of closed file
```

This may be surmountable with further edits to workunits, but I'm not sure how to achieve that right now.